src/components/FeatureViews/Scheduler/SessionView.js

```diff
@@ -31,36 +41,109 @@ class SessionView extends Component {
             "open": false,
             "current": "current",
             "all": "all",
+            "students": {
+                0: {
+                    name: "Calvin Fronda",
+                    id: 8,
+                },
+                1: {
+                    name: "Alex McGuire",
+                    id: 6,
+                },
+                2: {
+                    name: "Joe Buddy",
+                    id: 4,
+                },
+                3: {
+                    name: "Able Strong Body",
+                    id: 11,
+                }
+            }
         };
     }
 
     handleOpen = () => {
-        this.setState({"open": true});
+        this.setState({ "open": true });
     }
 
     handleClose = () => {
-        this.setState({"open": false});
+        this.setState({ "open": false });
     }
 
-    componentWillMount() {
-        this.setState(() => {
-            const sessionData = this.props.courseSessions[this.props.computedMatch.params.course_id][this.props.computedMatch.params.session_id];
-            const courseData = this.props.courses[this.props.computedMatch.params.course_id];
-            return {
-                sessionData,
-                courseData,
-            };
-        });
+    componentDidMount() {
+        this.props.calendarActions.fetchSessions({ id: this.props.match.params.session_id });
+        this.props.courseActions.fetchCourses(this.props.match.params.course_id);
+    }
+
+    componentDidUpdate(prevProps, prevState, snapshot) {
+        if (this.props !== prevProps &&
+            this.props.courseSessions.length !== 0 &&
+            Object.entries(this.props.courses).length !== 0) {
+            this.setState(() => {
+                const sessionData = this.props.courseSessions.find((session) => {
+                    return session.course === Number(this.props.match.params.course_id) &&
+                        session.id === Number(this.props.match.params.session_id);
```

Comment:
Suggest converting both to Number before doing the comparison, or converting both to String.